mopidy_musicbox_webclient/static/js/controls.js

```diff
@@ -338,74 +338,129 @@
+                    artists = artistsToString(track.album.artists)
+                }
+
+                if (artists.length > 0) {
+                    row.label = 'Artist'
+                    if (track.artists && track.artists.length > 1 || track.album && track.album.artists && track.album.artists.length > 1) {
+                        row.label += 's'
                     }
+                    row.text = artists
+                    html += stringFromTemplate(rowTemplate, row)
                 }
 
-                // Fallback to album artists.
-                if (artistNames.length === 0 && track.album && track.album.artists && track.album.artists.length > 0) {
-                    for (i = 0; i < track.album.artists.length; i++) {
-                        if (i > 0) {
-                            artistNames = artistNames + ', '
-                        }
-                        artistNames = artistNames + track.album.artists[i].name
+                var composers = artistsToString(track.composers)
+                if (composers.length > 0) {
+                    row.label = 'Composer'
+                    if (track.composers.length > 1) {
+                        row.label += 's'
                     }
+                    row.text = composers
+                    html += stringFromTemplate(rowTemplate, row)
                 }
-                if (artistNames.length > 0) {
-                    $('#popupShowInfo #artist-cell').text(artistNames)
-                    $('#popupShowInfo #artist-row').show()
-                } else {
-                    $('#popupShowInfo #artist-row').hide()
+
+                var performers = artistsToString(track.performers)
+                if (performers.length > 0) {
+                    row.label = 'Performer'
+                    if (track.performers.length > 1) {
+                        row.label += 's'
+                    }
+                    row.text = performers
+                    html += stringFromTemplate(rowTemplate, row)
                 }
+
+                if (track.genre) {
+                    row = {'label': 'Genre', 'text': track.genre}
+                    html += stringFromTemplate(rowTemplate, row)
+                }
+
                 if (track.track_no) {
-                    $('#popupShowInfo #track-no-cell').text(track.track_no)
-                    $('#popupShowInfo #track-no-row').show()
-                } else {
-                    $('#popupShowInfo #track-no-row').hide()
+                    row = {'label': 'Track #', 'text': track.track_no}
+                    html += stringFromTemplate(rowTemplate, row)
                 }
+
+                if (track.disc_no) {
+                    row = {'label': 'Disc #', 'text': track.disc_no}
+                    html += stringFromTemplate(rowTemplate, row)
+                }
+
+                if (track.date) {
+                    row = {'label': 'Date', 'text': new Date(track.date)}
```

Comment:
The easiest approach, which appears to be more or less standard and well supported, is to use ``localeString()``.